complex-numbers machine-precision friendly unittest

Unittest must be changed in order to take into account machine precision? or there is something wrong with my machine?

- __OS:__ _Ubuntu 16.04 x64_
- __Python:__ _3.5.2_

Two methods fail for me: __exp__ and __truediv__.
Here is my implementation of __Division__ and __Exponentiation__:
```
# Get cartesian representation from polar representation
    @staticmethod
    def cartesian_from_polar( r, theta ):
        """Given z = r * exp( i * theta ), calculates
        z = real + i * imaginary
        """
        # Cartesian Representation
        real = r * math.cos(theta)
        imaginary = r * math.sin(theta)
        # Return
        return( real, imaginary )
# Division
def __truediv__(self, other):
        """Complex Number Division
        z = r * exp( i * theta ), w = r' * exp( i * theta' ) =>
        z / w = ( r/r' ) * exp( i*(theta - theta') )
        """
        # Calculate Components
        r = self.r / other.r
        theta = self.theta - other.theta
        real, imaginary = ComplexNumber.cartesian_from_polar( r, theta )
        # Return new instance
        return( ComplexNumber( real, imaginary ) )
# Exponentiation
def exp(self):
        """Complex Number Exponentiation
        z = a + i * b =>
        exp(z) = exp(a) * exp( i * b )
        """
        # Calculate Components
        r = math.exp(self.real)
        theta = self.imaginary
        real, imaginary = ComplexNumber.cartesian_from_polar( r, theta )
        # Return new instance
        return( ComplexNumber( real, imaginary ) )
```
here is the failed unittest:
```
...........F..F.F..............
======================================================================
FAIL: test_divide_numbers_with_real_and_imaginary_part (__main__.ComplexNumbersTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "complex_numbers_test.py", line 114, in test_divide_numbers_with_real_and_imaginary_part
    self.assertEqual(input_number / divider, expected)
AssertionError: 0.44 +i*0.07999999999999999 != 0.44 +i*0.08

======================================================================
FAIL: test_eulers_identity_formula (__main__.ComplexNumbersTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "complex_numbers_test.py", line 156, in test_eulers_identity_formula
    self.assertEqual(input_number.exp().imaginary, expected.imaginary)
AssertionError: 1.2246467991473532e-16 != 0

======================================================================
FAIL: test_exponential_of_a_number_with_real_and_imaginary_part (__main__.ComplexNumbersTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "complex_numbers_test.py", line 174, in test_exponential_of_a_number_with_real_and_imaginary_part
    self.assertEqual(input_number.exp().imaginary, expected.imaginary)
AssertionError: 2.4492935982947064e-16 != 0
```

--- a/exercises/complex-numbers/complex_numbers_test.py
+++ b/exercises/complex-numbers/complex_numbers_test.py
@@ -152,26 +152,30 @@ class ComplexNumbersTest(unittest.TestCase):
     def test_eulers_identity_formula(self):
         input_number = ComplexNumber(0, math.pi)
         expected = ComplexNumber(-1, 0)
-        self.assertEqual(input_number.exp().real, expected.real)
-        self.assertEqual(input_number.exp().imaginary, expected.imaginary)
+        actual = input_number.exp()
+        self.assertAlmostEqual(actual.real, expected.real)
+        self.assertAlmostEqual(actual.imaginary, expected.imaginary)
 
     def test_exponential_of_0(self):
         input_number = ComplexNumber(0, 0)
         expected = ComplexNumber(1, 0)
-        self.assertEqual(input_number.exp().real, expected.real)
-        self.assertEqual(input_number.exp().imaginary, expected.imaginary)
+        actual = input_number.exp()
+        self.assertAlmostEqual(actual.real, expected.real)
+        self.assertAlmostEqual(actual.imaginary, expected.imaginary)
 
     def test_exponential_of_a_purely_real_number(self):
         input_number = ComplexNumber(1, 0)
         expected = ComplexNumber(math.e, 0)
-        self.assertEqual(input_number.exp().real, expected.real)
-        self.assertEqual(input_number.exp().imaginary, expected.imaginary)
+        actual = input_number.exp()
+        self.assertAlmostEqual(actual.real, expected.real)
+        self.assertAlmostEqual(actual.imaginary, expected.imaginary)
 
     def test_exponential_of_a_number_with_real_and_imaginary_part(self):
         input_number = ComplexNumber(math.log(2), math.pi)
         expected = ComplexNumber(-2, 0)
-        self.assertEqual(input_number.exp().real, expected.real)
-        self.assertEqual(input_number.exp().imaginary, expected.imaginary)
+        actual = input_number.exp()
+        self.assertAlmostEqual(actual.real, expected.real)
+        self.assertAlmostEqual(actual.imaginary, expected.imaginary)
 
 
 if __name__ == '__main__':

--- a/exercises/complex-numbers/example.py
+++ b/exercises/complex-numbers/example.py
@@ -40,6 +40,6 @@ class ComplexNumber(object):
         return ComplexNumber(self.real, -1 * self.imaginary)
 
     def exp(self):
-        r = round(math.cos(self.imaginary), 8) * math.exp(self.real)
-        i = round(math.sin(self.imaginary), 8) * math.exp(self.real)
+        r = math.cos(self.imaginary) * math.exp(self.real)
+        i = math.sin(self.imaginary) * math.exp(self.real)
         return ComplexNumber(r, i)

